src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.ValidateToken.Internal.cs

```diff
@@ -40,13 +37,31 @@ internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
                     StackFrames.TokenValidationParametersNull);
             }
 
-            var conditionsResult = ValidateConditions(samlToken, validationParameters, callContext);
+            validationParameters = await SamlTokenUtilities.PopulateValidationParametersWithCurrentConfigurationAsync(validationParameters, cancellationToken).ConfigureAwait(false);
+
+            var conditionsResult = ValidateConditions(
+                samlToken,
+                validationParameters,
+                callContext);
 
             if (!conditionsResult.IsValid)
             {
                 return conditionsResult.UnwrapError().AddStackFrame(new StackFrame(true));
             }
 
+            Task<ValidationResult<ValidatedIssuer>> validatedIssuerResult = validationParameters.IssuerValidatorAsync(
```

Comment:
This should be `ValidationResult<ValidatedIssuer> validatedIssuerResult = await validationParameters.IssuerValidatorAsync(...`

Reply:
Updated. Thanks @iNinja